app/views/stages/_fields.html.erb

```diff
@@ -26,19 +26,54 @@
   <% if ENV['DEPLOY_GROUP_FEATURE'] %>
     <div class="form-group">
       <%= form.label :name, 'Deploy Groups', class: 'col-lg-2 control-label' %>
-      <div class="col-lg-4">
-        <%= form.collection_select(
-          :deploy_group_ids, DeployGroup.order(:environment_id, :name).all, :id, :long_name,
-          {
-            selected: (@stage.deploy_group_ids if @stage && @stage.deploy_groups),
-            include_blank: false,
-            include_hidden: false
-          },
-          {
-            multiple: 'true',
-            class: 'form-control'
-          }
-        ) %>
+      <div class="col-lg-5">
+        <% if @environments.count > 0 %>
+          <table class="table table-condensed text-left deploy-groups">
+            <thead>
+              <tr>
+                <% @environments.each do |environment| %>
+                  <th>
+                    <label>
+                      <input
+                      type="checkbox"
+                      id="<%= "env_#{environment.id}_toggle_all" %>"
+                      class="env-toggle-all"
+                      data-target="<%= "#{environment.name}_checkbox" %>"/>
+                      <%= environment.name %>
+                    </label>
+                  </th>
+                <% end %>
+              </tr>
+            </thead>
+            <tbody>
+              <% @environments.map {|e| e.deploy_groups.count }.max.to_i.times do |index| %>
+                <tr>
+                  <% @environments.each do |environment| %>
+                    <% if environment.deploy_groups[index] %>
+                      <td>
+                        <label>
+                          <input
```

Comment:
I'd be inclined to use the FormTagHelper methods if straight up FormHelper is too complicated in this scenario, but the only real win is consistency I suppose.